### ISSUE TITLE
not -option is not working with MergeRequests.all

### ISSUE THREAD

**Issue by sampsasaarela** [jdalrymple/gitbeaker#2015]
**Description**

Based on docs https://docs.gitlab.com/ee/api/merge_requests.html#list-project-merge-requests it should be able to exclude merge requests by `author_id` for example, but this library throws an error.

**Steps to reproduce**

Try to exclude merger request by author id for example:

```
MergeRequests.all({
    groupId: 1234
    not: {
        author_id: id
    },
})
```

**Expected behaviour**

List merge requests without specified author id

**Actual behaviour**

Error thrown.

**Possible fixes**

I think `not` hash should be added to [AllMergeRequestsOptions](https://github.com/jdalrymple/gitbeaker/blob/master/packages/core/src/resources/MergeRequests.ts#L56)

**Comment by jdalrymple** [comment]
Can you share what the error is?

**Comment by sampsasaarela** [comment]
> Can you share what the error is?

I don't remember exactly and can't test right now but I would remember it was because `not` is not defined in [AllMergeRequestsOptions](https://github.com/jdalrymple/gitbeaker/blob/master/packages/core/src/resources/MergeRequests.ts#L56) interface.

**Comment by jdalrymple** [comment]
Ah a 400 im guessing

**Comment by jdalrymple** [comment]
@sampsasaarela yea thats def a problem, but wouldnt cause the error. Anyways i am able to recreate, looking into why its happening now

**Comment by jdalrymple** [comment]
:rocket: Issue was released in [`34.1.0`](https://github.com/jdalrymple/gitbeaker/releases/tag/34.1.0) :rocket:
